### DISCUSSION TITLE
Can we import a normal 3D Model into OpenJSCAD and make it Parametric inside it (OpenJSCAD)?

### DISCUSSION THREAD

**Discussion by VikasKumar12-GL** [jscad/OpenJSCAD.org#1285] [Q&A]
Hi all,
1. We have a use case, where in we have to import a normal 3D Model into OpenJSCAD and make it parametric, so that we can then open this 3D Parametric Model in ThreeJS based renderer and change/adjust this Model's dimensions based on its different Parametric options?

Thanks

**Comment by z3dev** [comment]
@VikasKumar12-GL welcome!

So... let's split this into a couple of areas to discuss.

First, JSCAD supports projects which can 'require' files that contain external formats, i.e. STL, DXF, etc. See https://openjscad.xyz/dokuwiki/doku.php?id=en:design_guide_anatomy

Second, the suggested project seems to include some kind of custom website, which will allow the design parameters to be adjusted. There are several examples of these projects, and you might want to discuss at the User Group.

Third, rendering the JSCAD geometries via ThreeJS is possible. If you really require ThreeJS then you will have to put together a few custom routines. Maybe @hrgdavor can assist in this area.

**Comment by hrgdavor** [comment]
@VikasKumar12-GL there is an effort to produce jscad related components to create different apps, and also a new version of jscad website. There the rendering is being separeted so different 3d engines can show jscad modeling results. https://github.com/hrgdavor/jscadui/ ... A new jscad web prototype is tested on this url. https://jscad.app and it uses threejs for rendering. source is at https://github.com/hrgdavor/jscadui/tree/main/apps/jscad-web and @platypii is the main driving force in it becoming a reality (progress was much slower while I was alone contributing there).  If you feel chatty you can also reach us on discord: https://discord.gg/AaqGskur93

**Reply by VikasKumar12-GL** [reply]
Thanks @hrgdavor , let me go through your response and the response from @z3dev, try to understand and run the above code locally, before I come up any other question. This is great information in deed.

**Comment by goyalyashpal** [comment]
> A new jscad web prototype is tested on this url. https://jscad.app and

hi! so, the jscad.app _is_ official too. Seeing it mentioned nowhere on the readme, i got a bit worried if that was a third party clone or smth (similar to this one: https://github.com/doublecmd/doublecmd.github.io/issues/52#issuecomment-1427060807)

jscad.app links i stumbled upon from web search:
* https://jscad.app/docs/tutorial-01_gettingStarted.html
* https://jscad.app
